src/sinks/datadog/logs/sink.rs

```diff
@@ -102,6 +103,50 @@ fn normalize_event(event: &mut Event) {
         log.rename_key(host_path, event_path!("hostname"));
     }
 
+    if let Some(ddtags_path) = log.find_key_by_meaning("tags") {
+        let ddtags_path = ddtags_path.clone();
+
+        // first, if the value is an array we need to reconstruct it to a comma delimited string for DD logs intake.
+        if let Some(Value::Array(tags_arr)) = log.get(&ddtags_path) {
+            let n_tags = tags_arr.len();
+            if n_tags > 0 {
+                let mut all_tags = String::new();
+                tags_arr.iter().enumerate().for_each(|(idx, tag_kv)| {
+                    if tag_kv.is_bytes() {
+                        let tag_kv_str = String::from_utf8_lossy(
+                            tag_kv.as_bytes().expect("tag should be bytes."),
+                        );
+                        all_tags.push_str(&tag_kv_str);
+                        if idx + 1 < n_tags {
+                            all_tags.push(',');
+                        }
+                    }
+                });
+                log.insert(&ddtags_path, all_tags);
+            }
+        }
+
+        // now, the tags attribute must be at the event root so we will move it there if
+        // needed and move any conflicting field if any.
+        if ddtags_path != parse_target_path("tags").expect("should parse tags target path") {
```

Comment:
Is there not a macro to produce a target path from a constant? If not, you should be able to pre-parse this and store it somewhere else rather than parsing on every event.

Reply:
In the split off PR for this, I used a different approach, which doesn't require creating this target path for comparison.